yt/yt/library/decimal/decimal.h

```diff
@@ -5,29 +5,43 @@
 #include <util/system/defaults.h>
 #include <util/generic/string.h>
 
+#include <array>
+
 namespace NYT::NDecimal {
 
 ////////////////////////////////////////////////////////////////////////////////
 
 class TDecimal
 {
 public:
+    //! Both types defined below represent signed values. They are only intended
+    //! to be used for manipulating binary data, since they don't actually expose
+    //! any arithmetic operations. You should avoid dealing with individual parts.
     struct TValue128
     {
         ui64 Low;
         i64 High;
     };
     static_assert(sizeof(TValue128) == 2 * sizeof(ui64));
 
+    struct TValue256
+    {
+        std::array<ui32, 8> Parts;
```

Comment:
1. So that we can use ui64 for implementing arithmetic operations. 
2. I didn't have any such plans, we are quite dependent on them being binary representations of the corresponding integers in various interop, so it won't be convenient to use the same type. We will still need some `memcpy128`, memcpy256` like operations.